src/main/java/com/ankoma88/datastructures/Performance/service/PerformanceAnalysisService.java

```diff
@@ -0,0 +1,186 @@
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executor;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import static com.ankoma88.datastructures.performance.domain.model.DataStructureEnum.ARRAYLIST;
+import static com.ankoma88.datastructures.performance.domain.model.DataStructureEnum.LINKEDLIST;
+import static com.ankoma88.datastructures.performance.domain.model.OperationEnum.*;
+import static com.ankoma88.datastructures.performance.domain.model.ScenarioEnum.*;
+
+@Service
+public class PerformanceAnalysisService implements AnalysisService {
+
+    private static final Logger LOGGER = Logger.getLogger(PerformanceAnalysisService.class.getName());
+
+    @Autowired
+    @Qualifier("measurementsTaskExecutor")
+    private Executor measurementsThreadPool;
+
+    @Autowired
+    private MeasurementsRepository measurementsRepository;
+
+    @Override
+    public List<PerformanceMeasurementDto> getLatestMeasurements(int size){
+        PageRequest pageRequest = PageRequest.of(0, size);
+        Page<MeasurementRecord> records = measurementsRepository.getMeasurementRecords(pageRequest);
+        List<MeasurementRecord> measurementRecords = records.getContent();
+        return buildMeasurements(measurementRecords);
+    }
+
+    @Override
+    public PerformanceDto measureOperations(int listSize, int repeats) {
+
+        final CompletableFuture<MeasurementDto> arrayListCreateCF = buildCompletableFuture(listSize, repeats, CREATE, ARRAYLIST);
+        final CompletableFuture<MeasurementDto> arrayListReadCF = buildCompletableFuture(listSize, repeats, READ, ARRAYLIST);
+        final CompletableFuture<MeasurementDto> arrayListUpdateCF = buildCompletableFuture(listSize, repeats, UPDATE, ARRAYLIST);
+        final CompletableFuture<MeasurementDto> arrayListDeleteCF = buildCompletableFuture(listSize, repeats, DELETE, ARRAYLIST);
+
+        final CompletableFuture<MeasurementDto> linkedListCreateCF = buildCompletableFuture(listSize, repeats, CREATE, LINKEDLIST);
+        final CompletableFuture<MeasurementDto> linkedListReadCF = buildCompletableFuture(listSize, repeats, READ, LINKEDLIST);
+        final CompletableFuture<MeasurementDto> linkedListUpdateCF = buildCompletableFuture(listSize, repeats, UPDATE, LINKEDLIST);
+        final CompletableFuture<MeasurementDto> linkedListDeleteCF = buildCompletableFuture(listSize, repeats, DELETE, LINKEDLIST);
+
+        AnalysisDto arrayListAnalysis = null;
+        AnalysisDto linkedListAnalysis = null;
+        PerformanceDto performance;
+
+        try {
+            arrayListAnalysis = new AnalysisDto(
+                    arrayListCreateCF.get(),
+                    arrayListReadCF.get(),
+                    arrayListUpdateCF.get(),
+                    arrayListDeleteCF.get()
+            );
+        } catch (InterruptedException e) {
```

Comment:
Repeated exception handling logic for InterruptedException and ExecutionException. Extract common exception handling into a separate method to avoid code duplication.